resolvers/ambulance.js

```diff
@@ -68,20 +68,27 @@ const ambulanceResolvers = {
                 args.id
             );
           }
-        });
-
-      await db.eventAmbulances.destroy({
-        where: {
-          ambulanceId: args.id,
-        },
-      });
-
-      return db.ambulance.destroy({
-        where: {
-          id: args.id,
-        },
-      });
-    },
+        })
+        .then(() =>
+          db.eventAmbulances.destroy({
+            where: {
+              ambulanceId: args.id,
+            },
+          })
+        )
+        .then(() =>
+          db.ambulance.destroy({
+            where: {
+              id: args.id,
+            },
+          })
+        )
+        .then((isDeleted) => {
+          if (isDeleted === 1) {
+            return args.id;
+          }
+          throw new Error('Deletion failed for ambulance ID: ' + args.id);
+        }),
```

Comment:
`destroy` returns a Promise that always gets resolved with `1` on success or `0` otherwise (i.e. the promise is always “fulfilled” and won’t trigger a `catch` statement event when the deleting failed)